components/templates/src/global_fns/mod.rs

```diff
@@ -140,6 +142,39 @@ impl TeraFn for ResizeImage {
     }
 }
 
+#[derive(Debug)]
+pub struct GetImageMeta {
+    content_path: PathBuf
+}
+
+impl GetImageMeta {
+    pub fn new(content_path: PathBuf) -> Self {
+        Self { content_path }
+    }
+}
+
+impl TeraFn for GetImageMeta {
+    fn call(&self, args: &HashMap<String, Value>) -> Result<Value> {
+        let path = required_arg!(
+            String,
+            args.get("path"),
+            "`get_image_meta` requires a `path` argument with a string value"
+        );
+        let src_path = self.content_path.join(&path);
+        if !src_path.exists(){
+            return Err(format!("`get_image_meta`: Cannot find path: {}", path).into());
+        }
+        let img = image::open(&src_path)
+            .map_err(|e| Error::chain(format!("Failed to process image: {}", path), e))?;
```

Comment:
Failed to process image -> Failed to open image maybe? This doesn't do any processing